lib/elements/clone.py

```diff
@@ -67,81 +59,84 @@ def clone_fill_angle(self):
+
+                # Strip out the translation component from the matrix
+                t = Transform((t.a, t.b, t.c, t.d, 0, 0))
+                try:
+                    return t.rotation_degrees()
+                except ValueError:
+                    # Something is weird about the matrix such that it apparently isn't a rotation with a uniform scale.
+                    # So, next best thing is to apply it to a unit vector and pull the angle from that.
+                    return degrees(t.apply_to_point((1, 0)).angle)
+
+            source_angle = angle_for_transform(source_transform)
+            cloned_angle = angle_for_transform(cloned_transform)
+            angle = cloned_angle - source_angle
+
+        elements = self.clone_to_elements(cloned_node)
         for element in elements:
-            stitch_groups = element.to_stitch_groups(last_patch)
-            patches.extend(stitch_groups)
-
-        source_node.set('transform', old_transform)
-        source_node.set(INKSTITCH_ATTRIBS['angle'], old_angle)
-        return patches
+            # We manipulate the element's node directly here instead of using get/set param methods, because otherwise
+            # we run into issues due to those methods' use of caching not updating if the underlying param value is changed.
 
-    def _get_rotation(self, transform, source_node, source_path):
-        try:
-            rotation = transform.rotation_degrees()
-        except ValueError:
-            source_path = CubicSuperPath(source_path)[0]
-            clone_path = Path(source_node.get_path()).transform(source_node.composed_transform())
-            clone_path = CubicSuperPath(clone_path)[0]
-
-            angle_source = atan2(source_path[1][1][1] - source_path[0][1][1], source_path[1][1][0] - source_path[0][1][0])
-            angle_clone = atan2(clone_path[1][1][1] - clone_path[0][1][1], clone_path[1][1][0] - clone_path[0][1][0])
-            angle_embroidery = radians(-float(source_node.get(INKSTITCH_ATTRIBS['angle'], 0)))
-
-            diff = angle_source - angle_embroidery
-            rotation = degrees(diff + angle_clone)
+            if self.clone_fill_angle is None:  # Normally, rotate the cloned element's angle by the clone's rotation.
+                element_angle = float(element.node.get(INKSTITCH_ATTRIBS['angle'], 0)) - angle
+            else:  # If clone_fill_angle is specified, override the angle instead.
+                element_angle = self.clone_fill_angle
 
             if self.flip_angle:
-                rotation = -degrees(diff - angle_clone)
+                element_angle = -element_angle
 
-        return -rotation
+            element.node.set(INKSTITCH_ATTRIBS['angle'], element_angle)
 
-    def get_clone_style(self, style_name, node, default=None):
-        style = node.style[style_name] or default
-        return style
+            stitch_groups = element.to_stitch_groups(last_patch)
```

Comment:
Since we are possibly looping through several elements now, we can set the last stitch group (if existent) as the last_patch and pass it over, so that the next element may start at a better spot. last_patch can be renamed to `last_stitch_group` as well.

Reply:
Oh, duh, that's an obvious thing to overlook. Easy enough.